svn: can not be found

Hi folks,

When running the GitHub action to deploy to WordPress SVN (which has never had an issue), I'm now seeing the following:

`/home/runner/work/_actions/10up/action-wordpress-plugin-deploy/stable/deploy.sh: line 6[3](https://github.com/alicolville/shortcode-variables/actions/runs/12412813078/job/34657268003#step:3:3): svn: command not found`

But I'm not quite sure how to address this, any tips would be much appreciated!
```
##[debug]Evaluating: secrets.SVN_PASSWORD
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'SVN_PASSWORD'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Evaluating: secrets.SVN_USERNAME
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'SVN_USERNAME'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Evaluating condition for step: 'WordPress Plugin Deploy'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: WordPress Plugin Deploy
##[debug]Loading inputs
##[debug]Loading env
Run 10up/action-wordpress-plugin-deploy@stable
##[debug]Evaluating: inputs.generate-zip
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'generate-zip'
##[debug]=> 'false'
##[debug]Result: 'false'
##[debug]Evaluating: inputs.dry-run
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'dry-run'
##[debug]=> 'false'
##[debug]Result: 'false'
##[debug]Evaluating condition for step: 'run'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: run
##[debug]Loading inputs
##[debug]Evaluating: format('{0}/deploy.sh', github.action_path)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '{0}/deploy.sh'
##[debug]..Evaluating Index:
##[debug]....Evaluating github:
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'action_path'
##[debug]..=> '/home/runner/work/_actions/10up/action-wordpress-plugin-deploy/stable'
##[debug]=> '/home/runner/work/_actions/10up/action-wordpress-plugin-deploy/stable/deploy.sh'
##[debug]Result: '/home/runner/work/_actions/10up/action-wordpress-plugin-deploy/stable/deploy.sh'
##[debug]Loading env
Run /home/runner/work/_actions/10up/action-wordpress-plugin-deploy/stable/deploy.sh
##[debug]/usr/bin/bash --noprofile --norc -e -o pipefail /home/runner/work/_temp/5506b3dd-c7f1-4843-ac35-59d3c10b[2](https://github.com/alicolville/shortcode-variables/actions/runs/12412813078/job/34657268003#step:3:2)3d4.sh
allexport      	off
braceexpand    	on
emacs          	off
errexit        	on
errtrace       	off
functrace      	off
hashall        	on
histexpand     	off
history        	off
ignoreeof      	off
interactive-comments	on
keyword        	off
/home/runner/work/_actions/10up/action-wordpress-plugin-deploy/stable/deploy.sh: line 6[3](https://github.com/alicolville/shortcode-variables/actions/runs/12412813078/job/34657268003#step:3:3): svn: command not found
monitor        	off
noclobber      	off
noexec         	off
noglob         	off
nolog          	off
notify         	off
nounset        	off
onecmd         	off
physical       	off
pipefail       	off
posix          	off
privileged     	off
verbose        	off
vi             	off
xtrace         	off
ℹ︎ SLUG is shortcode-variables
ℹ︎ VERSION is [4](https://github.com/alicolville/shortcode-variables/actions/runs/12412813078/job/34657268003#step:3:4).2
ℹ︎ ASSETS_DIR is .wordpress-org
➤ Checking out .org repository...
Error: Process completed with exit code 12[7](https://github.com/alicolville/shortcode-variables/actions/runs/12412813078/job/34657268003#step:3:7).
##[debug]Finished: run
##[debug]Evaluating: steps.deploy.outputs.zip-path
##[debug]Evaluating Index:
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating steps:
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'deploy'
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'outputs'
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'zip-path'
##[debug]=> null
##[debug]Result: null
##[debug]Finishing: WordPress Plugin Deploy
```

--- a/deploy.sh
+++ b/deploy.sh
@@ -6,6 +6,32 @@
 # it does not exit with 0, as we are interested in the final exit.
 set -eo
 
+# Function to check if a command exists
+command_exists() {
+	command -v "$1" >/dev/null 2>&1
+}
+
+# Check if SVN is installed
+if command_exists svn; then
+	echo "SVN is already installed."
+else
+	echo "SVN is not installed. Installing SVN..."
+
+	# Update the package list
+	sudo apt-get update -y
+
+	# Install SVN
+	sudo apt-get install -y subversion
+
+	# Verify installation
+	if command_exists svn; then
+		echo "SVN was successfully installed."
+	else
+		echo "Failed to install SVN. Please check your system configuration."
+		exit 1
+	fi
+fi
+
 # Ensure SVN username and password are set
 # IMPORTANT: while secrets are encrypted and not viewable in the GitHub UI,
 # they are by necessity provided as plaintext in the context of the Action,
@@ -44,12 +70,12 @@ echo "ℹ︎ ASSETS_DIR is $ASSETS_DIR"
 
 if [[ -z "$BUILD_DIR" ]] || [[ $BUILD_DIR == "./" ]]; then
 	BUILD_DIR=false
-elif [[ $BUILD_DIR == ./* ]]; then 
+elif [[ $BUILD_DIR == ./* ]]; then
 	BUILD_DIR=${BUILD_DIR:2}
 fi
 
 if [[ "$BUILD_DIR" != false ]]; then
-	if [[ $BUILD_DIR != /* ]]; then 
+	if [[ $BUILD_DIR != /* ]]; then
 		BUILD_DIR="${GITHUB_WORKSPACE%/}/${BUILD_DIR%/}"
 	fi
 	echo "ℹ︎ BUILD_DIR is $BUILD_DIR"

--- a/deploy.sh
+++ b/deploy.sh
@@ -140,7 +140,7 @@ if [[ "$BUILD_DIR" = false ]]; then
 		git config --global user.name "10upbot on GitHub"
 
 		# Ensure git archive will pick up any changed files in the directory try.
-		test $(git ls-files --deleted) && git rm $(git ls-files --deleted)
+		test "$(git ls-files --deleted)" && git rm "$(git ls-files --deleted)"
 		if [ -n "$(git status --porcelain --untracked-files=all)" ]; then
 			git add .
 			git commit -m "Include build step changes"

--- a/examples/deploy-on-publishing-a-new-release-and-attach-a-zip-file-to-the-release.yml
+++ b/examples/deploy-on-publishing-a-new-release-and-attach-a-zip-file-to-the-release.yml
@@ -55,11 +55,6 @@ jobs:
         npm install
         npm run build
 
-    - name: Install SVN ( Subversion )
-      run: |
-        sudo apt-get update
-        sudo apt-get install subversion
-
     - name: WordPress Plugin Deploy
 
       # You can add unique ids to specific steps if you want to reference their output later in the workflow.

--- a/examples/deploy-on-pushing-a-new-tag-and-create-release-with-attached-zip.yml
+++ b/examples/deploy-on-pushing-a-new-tag-and-create-release-with-attached-zip.yml
@@ -14,10 +14,6 @@ jobs:
       run: |
         npm install
         npm run build
-    - name: Install SVN ( Subversion )
-      run: |
-        sudo apt-get update
-        sudo apt-get install subversion
     - name: WordPress Plugin Deploy
       id: deploy
       uses: 10up/action-wordpress-plugin-deploy@stable
@@ -32,4 +28,4 @@ jobs:
         files: ${{github.workspace}}/${{ github.event.repository.name }}.zip
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        
+

--- a/examples/deploy-on-pushing-a-new-tag.yml
+++ b/examples/deploy-on-pushing-a-new-tag.yml
@@ -13,14 +13,10 @@ jobs:
       run: |
         npm install
         npm run build
-    - name: Install SVN ( Subversion )
-      run: |
-        sudo apt-get update
-        sudo apt-get install subversion
     - name: WordPress Plugin Deploy
       uses: 10up/action-wordpress-plugin-deploy@stable
       env:
         SVN_PASSWORD: ${{ secrets.SVN_PASSWORD }}
         SVN_USERNAME: ${{ secrets.SVN_USERNAME }}
         SLUG: my-super-cool-plugin # optional, remove if GitHub repo name matches SVN slug, including capitalization
-        
+

